api/v1alpha3/virtualmachine_conversion.go

```diff
@@ -5,21 +5,70 @@
 package v1alpha3
 
 import (
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
 	ctrlconversion "sigs.k8s.io/controller-runtime/pkg/conversion"
 
+	"github.com/vmware-tanzu/vm-operator/api/utilconversion"
 	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha4"
 )
 
+func Convert_v1alpha4_VirtualMachineSpec_To_v1alpha3_VirtualMachineSpec(
+	in *vmopv1.VirtualMachineSpec, out *VirtualMachineSpec, s apiconversion.Scope) error {
+
+	if err := autoConvert_v1alpha4_VirtualMachineSpec_To_v1alpha3_VirtualMachineSpec(in, out, s); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func Convert_v1alpha4_VirtualMachineStatus_To_v1alpha3_VirtualMachineStatus(
+	in *vmopv1.VirtualMachineStatus, out *VirtualMachineStatus, s apiconversion.Scope) error {
+
+	if err := autoConvert_v1alpha4_VirtualMachineStatus_To_v1alpha3_VirtualMachineStatus(in, out, s); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func restore_v1alpha4_VirtualMachinePromoteDisksMode(dst, src *vmopv1.VirtualMachine) {
+	dst.Spec.PromoteDisksMode = src.Spec.PromoteDisksMode
+}
+
 // ConvertTo converts this VirtualMachine to the Hub version.
 func (src *VirtualMachine) ConvertTo(dstRaw ctrlconversion.Hub) error {
 	dst := dstRaw.(*vmopv1.VirtualMachine)
-	return Convert_v1alpha3_VirtualMachine_To_v1alpha4_VirtualMachine(src, dst, nil)
+	if err := Convert_v1alpha3_VirtualMachine_To_v1alpha4_VirtualMachine(src, dst, nil); err != nil {
+		return err
+	}
+
+	// Manually restore data.
+	restored := &vmopv1.VirtualMachine{}
+	if ok, err := utilconversion.UnmarshalData(src, restored); err != nil || !ok {
+		return err
+	}
+
+	// BEGIN RESTORE
+
+	restore_v1alpha4_VirtualMachinePromoteDisksMode(dst, restored)
```

Comment:
Don't we need to do this restore v1a1 and v1a2 as well?

Reply:
Ah yes, adding those. Adding v1alpha3/virtualmachine_conversion_test.go, which is copy-n-paste of   v1alpha2/virtualmachine_conversion_test.go's `VirtualMachine hub-spoke-hub` test, with `PromoteDisksMode` added to the spec. Not sure if I should prune some of the copied `hub` fields or leave as-is?

Reply:
Rebasing with Sai's change merged.. I will prune that to just PromoteDisksMode